### ISSUE TITLE
Kodi 17 question.

### ISSUE THREAD

**Issue by lorderr** [asciidisco/plugin.video.netflix#25]
Hi.

Im trying to get your plugin to work on kodi 17.

What I have got working on Linux (amd64)
- login and browsing (had to add python cypto dep manually. removed it as plugin dep)
- got inputstream adaptive to work
- added libwidevine.so.

For some reason I get audio only. But apart from everything seems well. Libadaptive seems to complain about; Decrypt Sample returns failure.

Do you know if the reason for not getting video to work is due to differences in kodi17 and agile or if it is a libwidevine thing? Im having some difficulties with finding the problem

**Comment by asciidisco** [comment]
@lorderr I´m sorry to tell you that, but it´s impossible to run the plugin on Kodi 17 (like in, it will never work with Kodi 17).

The problem is, that Kodi 17 only supports decrypting via Widevine. But Netlfix (like Amazon these days) follows a "decrypt and decode" DRM policy. Unfortunately inputstream.adaptive and Kodi 17 did not support decoding by Widevine. They´ve changed that in Kodi-Agile  (and the inputstream.adaptive agile branch). That´s why it also isn't possible with the Kodi 18 mainline project to use the plugin (which will change, as changes in Agile will be merged back one day).

TL;DR:  With Kodi Agile it´s possible to let Widevine decode the Video, no other Kodi version can do that. Changes will not be backported to Kodi 17, so this will never work with Kodi 17.

**Comment by RedSoxFan04** [comment]
That’s rather unfortunate. I was really hoping to be able to use plugin.video.netflix on my first generation Apple TV - yes, it still works. I can’t get it to update beyond Kodi 17 - and yes, I tried fsck. Besides, it already runs a slowly with Kodi 17. By the way, I’m running OSMC on my first generation Apple T.V.  If you’re unfamiliar with OSMC, it’s an OS which is a combination of Debian and Kodi, and it’s made for set-top boxes. I installed OSMC on my first generation Apple TV so I could have it play my non-iTunes videos without me having to re-encode all of them.

**Comment by asciidisco** [comment]
@RedSoxFan04 As already said, there is nothing we can do about it. It depends on changes done in the Kodi source, that can't be easily & therefor won't be backported to Kodi 17.
I´d say, ask the OSMC maintainers if an upgrade can be done once the changes are merged into the mainline branch.

**Comment by bam80** [comment]
Are the changes merged into the mainline already?

**Comment by RedSoxFan04** [comment]
My Kodi 17 issue has been resolved,. I managed to get my first generation Apple TV to update to Kodi 17  back in December. I'm sorry I didn't get back here to mention it. By the time I finally managed to get my first generation Apple TV to update, I had forgotten all about this plugin and also this "issue thread".

**Comment by shtrom** [comment]
Could this plugin help? https://github.com/aussieaddons/script.module.drmhelper It seems (from my 5 minutes reading, trying to fix a similar problem) to be downloading libwidevine on behalf of other plugins that need it.

**Comment by shtrom** [comment]
Would this plugin help? https://github.com/aussieaddons/script.module.drmhelper From my reading of https://github.com/aussieaddons/plugin.video.abc_iview/blob/master/resources/lib/play.py#L51-L64, it seems to be doing the necessary setup to up widevine, even on Kodi 17.

**Comment by mskrebs2020** [comment]
Hi,  I am interested in getting a display of data from the Netflix manifest (MPD) using this implementation.
Is it possible to get a display of the Netflix DASH manifest when running Netflix of the Kodi player?
Really I am just interested in obtaining video or audio parameters...encoding rate, resolution/display size, sample rate..for the streaming media.
